lua/no-neck-pain/util/color.lua

```diff
@@ -53,41 +54,49 @@ function C.init(win, name, backgroundColor)
     end
 
     backgroundColor = backgroundColor or defaultBackground
+    textColor = textColor or backgroundColor
 
     D.log(
         "Color.init",
         "groupName `%s` - window `%s` - backgroundColor `%s`",
-        groupName,
+        backgroundGroup,
         win,
         backgroundColor
     )
 
-    vim.cmd(string.format("highlight! clear %s NONE", groupName))
+    D.log("Color.init", "groupName `%s` - window `%s` - textColor `%s`", textGroup, win, textColor)
+
+    vim.cmd(string.format("highlight! clear %s NONE", backgroundGroup))
+
+    -- create group for background
     vim.cmd(
         string.format(
             "highlight! %s guifg=%s guibg=%s",
-            groupName,
+            backgroundGroup,
             backgroundColor,
```

Comment:
maybe one group is enough if we replace this value by the `textColor`, no?

```suggestion
            textColor,
```

Reply:
Don't we need two groups because of the way colors get applied at the end? For the `Normal` and `NormalNC` we want the custom text colors but not for the other highlight groups. If we had a single group, line numbers etc. would also get the text color, which was not my intention.

Reply:
Ah yes indeed that make sense!